diff --git a/http_test.ts b/http_test.ts
--- a/http_test.ts
+++ b/http_test.ts
@@ -221,4 +221,5 @@
 async function runTestCase(testCase: TestCase, host: string, port: number): Promise<boolean> {
     try {
         const conn = await Deno.connect({ hostname: host, port });
+        conn.unref();
 
@@ -260,2 +261,3 @@
                     );
+                    conn.close();
                     return resolve(isSuccess);
