--- src/quic/engine.zig
+++ src/quic/engine.zig
@@ -29,6 +29,7 @@
         if (engine.udp_fd != -1) {
             const dest_addr = spec.dest_sa;
             const sa_len: u32 = if (dest_addr.*.sa_family == std.posix.AF.INET6) @sizeOf(std.posix.sockaddr.in6) else @sizeOf(std.posix.sockaddr.in);
+            std.debug.print("quic sending packet of {} bytes to {}\n", .{datagram.len, dest_addr.*.sa_family});
             _ = std.os.linux.sendto(engine.udp_fd, datagram.ptr, datagram.len, 0, @ptrCast(dest_addr), sa_len);
         }

@@ -107,6 +108,7 @@
         }

         // feed packet into engine -> if lsquic needs to ack, it automatically triggers on_packets_out
+        std.debug.print("lsquic feeding {} bytes\n", .{data.len});
         _ = c.lsquic_engine_packet_in(self.engine_ptr, data.ptr, data.len, @ptrCast(local_sa), @ptrCast(peer_sa), @ptrCast(self), 0);

         // wake up the engine to flush its queue
