--- src/quic/engine.zig
+++ src/quic/engine.zig
@@ -48,6 +48,7 @@
         const engine = try std.heap.c_allocator.create(QuicEngine);
         engine.* = QuicEngine{ .ssl_ctx = ssl_ctx };

+        _ = c.lsquic_log_to_fstream(c.stderr, c.LLTS_NONE);
         // configure stream interface callbacks
         var stream_if = std.mem.zeroes(c.lsquic_stream_if);
         stream_if.on_new_stream = lsquic_api.on_new_stream;
